torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -820,13 +909,76 @@ def auto_functionalized_v2_fake(
 @auto_functionalized_v2.py_impl(ProxyTorchDispatchMode)
 def auto_functionalized_v2_proxy(
     mode,
-    _mutable_op: OpOverload,
-    **kwargs: dict[str, Any],
+    _mutable_op: _MutableOpType,
+    **kwargs: Any,
 ) -> tuple[Any, tuple[Tensor, ...]]:
+    if isinstance(_mutable_op, HigherOrderOperator):
+
+        def _maybe_materialize_callable_args(
```

Comment:
is it possible to avoid making this a nested function? it impacts readability

Reply:
sounds good!